vault/credentialprocessprovider.go

```diff
@@ -0,0 +1,53 @@
+package vault
+
+import (
+	"encoding/json"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+type CredentialProcessProvider struct {
+	CredentialProcess string
+	ExpiryWindow      time.Duration
+	credentials.Expiry
+}
+
+type CredentialProcessResponse struct {
+	AccessKeyId     string    `json:"AccessKeyId"`
+	SecretAccessKey string    `json:"SecretAccessKey"`
+	SessionToken    string    `json:"SessionToken"`
+	Expiration      time.Time `json:"Expiration"`
+	Version         int       `json:"Version"`
+}
+
+// Retrieve fetch credentials from an external process
+func (p *CredentialProcessProvider) Retrieve() (credentials.Value, error) {
+	cred, err := p.callCredentialProcess()
+	if err != nil {
+		return credentials.Value{}, err
+	}
+
+	p.SetExpiration(cred.Expiration, p.ExpiryWindow)
+	return credentials.Value{
+		AccessKeyID:     cred.AccessKeyId,
+		SecretAccessKey: cred.SecretAccessKey,
+		SessionToken:    cred.SessionToken,
+	}, nil
+}
+
+func (p *CredentialProcessProvider) callCredentialProcess() (CredentialProcessResponse, error) {
+	params := strings.Split(p.CredentialProcess, " ")
```

Comment:
You'll want to either use [shellwords](https://github.com/Wing924/shellwords) here or pass it to an actual shell as [AssumeRoleWithWebIdentityProvider does](https://github.com/99designs/aws-vault/pull/587/files#diff-cf36306ae347a936c254fd164fd396ceR93-R97)

Reply:
Updated based on AssumeRoleWithWebIdentityProvider as suggested